packages/keychain/src/utils/account.ts

```diff
@@ -133,78 +140,36 @@ class Account extends BaseAccount {
     });
 
     try {
-      if (
-        this.status === Status.COUNTERFACTUAL ||
-        this.status === Status.DEPLOYING
-      ) {
-        const registerTxnHash = Storage.get(this.selector).registerTxnHashHash;
-        if (registerTxnHash) {
-          this.status = Status.REGISTERING;
-          this.rpc
-            .waitForTransaction(
-              registerTxnHash,
-              Account.waitForTransactionOptions,
-            )
-            .then(() => this.sync());
-          return;
-        }
-
-        const data = await this.getContract();
-        // @ts-expect-error TODO: fix type error
-        if (!data?.contract?.deployTransaction?.id) {
-          return;
-        }
-
-        // @ts-expect-error TODO: fix type error
-        const deployTxnHash = data.contract.deployTransaction.id.split("/")[1];
-        const deployTxnReceipt = await this.rpc.getTransactionReceipt(
-          deployTxnHash,
-        );
-
-        // Pending txn so poll for inclusion.
-        if (!("execution_status" in deployTxnReceipt)) {
-          this.status = Status.DEPLOYING;
-
-          if (!this.waitingForDeploy) {
-            this.rpc
-              .waitForTransaction(
-                deployTxnHash,
-                Account.waitForTransactionOptions,
-              )
-              .then(() => this.sync());
-            this.waitingForDeploy = true;
+      switch (this.status) {
+        case Status.DEPLOYING:
+        case Status.COUNTERFACTUAL:
+          const hash = await this.getDeploymentTxn();
+          const receipt = await this.rpc.getTransactionReceipt(hash);
+          if (receipt.isRejected()) {
+            // TODO: Handle redeployment
+            this.status = Status.REJECTED;
+            return;
```

Comment:
It's usually a good practice to wrap each case in expression so each const is scoped.

```
case XXX: {
  const hash = await this.getDeploymentTxn();
  // ...
  break;
}